### DISCUSSION TITLE
How does someone build and run this locally like in enclave?

### DISCUSSION THREAD

**Discussion by jsdev** [material-components/material-web#5545] [Q&A] [answered]
### Description

Could you please add build instructions for noobs.

### Browser/OS Environment

Not Browser/OS specific this the command s needed to run locally like what comes after npm run build script.

is it node server.js?

**Comment by asyncliz** [comment]
We have a [section on building](https://github.com/material-components/material-web/blob/main/docs/quick-start.md#building) in our quick start guide. It links out to [lit.dev's modern build for production](https://lit.dev/docs/tools/production/#modern-only-build) for a deeper dive on how to create a bundle JavaScript file.

You can include that file with an HTML script tag, or elsewhere in your app's build system. Does that help out?

**Comment by jsdev** [comment]
Thanks for response. 
The section and links you provided are not specific to the repo they are in general to a creating something new with material web components.

I am asking how to build material-web.dev basically on localhost, there doesn't seem to be index.js or index.html like how does it know how to create the nav-drawer on the left etc. I am struggling to see how I can build locally and aspire to make a pull request contribution.

I'm used to opensource being something someone can pull down locally and learn by tinkering, and everything needed to run just being a command in the package.json.

This repo doesn't have a rollup.config.js, or index.js checked in, which is necessary in the documentation.

**Reply by asyncliz** [reply, marked as answer]
You can use the following command to [run the catalog's dev server](https://github.com/material-components/material-web/blob/6f354ac85c52a490b32edfbc05a2a70eb5c9402b/catalog/package.json#L10). You need to run the command while inside the `./catalog` directory.

```sh
cd ./catalog
npm start
```

We don't have docs on the catalog since the docs are for the components that folks can use in the library, not our catalog server which isn't published to npm.

The catalog's build files are located [in the catalog folder](https://github.com/material-components/material-web/tree/main/catalog) (eleventy, esbuild, and firebase).
